### PR TITLE
Workflow executor configurations

### PR DESCRIPTION
**Pull request by Meruja**
Reference: https://apim.docs.wso2.com/en/latest/learn/design-api/advanced-topics/adding-an-api-state-change-workflow/

### CODE REVIEW

File: en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md [hunk 1]
```diff
@@ -62,8 +62,10 @@ First, enable the API state change workflow for **Approval Workflow Executor**.
...
-| `stateList`| Mandatory          | This is a comma separated list of the current state and intended action. For example, Created:Publish,Published:Block 
-
-                                                                |
+| Element Name               | Mandatory/Optional | Description                                                                                                                                                                                         |
+|----------------------------|--------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `processDefinitionKey`     | Mandatory          | BPMN process definition ID.  BPMN process provided with AM as default has `APIStateChangeApprovalProcess` as the ID                                                                                 |
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md:67]: I don't see this in the configuration.  where is it?

File: en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md [hunk 2]
```diff
@@ -62,8 +62,10 @@ First, enable the API state change workflow for **Approval Workflow Executor**.
...
+| `stateList`                | Mandatory          | This is a comma separated list of the current state and intended action. For example, Created:Publish,Published:Block                                                                               |
+| `serviceEndpoint`          | Optional           | The URL of the BPMN process engine. This overrides the global `<ServerUrl>` value from the `api-manager.xml` file. This can be used to connect a separate workflow engine for a tenant.             |
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md:69]: What we have currently is the inbuilt Workflow process. This document explains that. In this case, these configurations do not matter right?

File: en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md [hunk 3]
```diff
@@ -62,8 +62,10 @@ First, enable the API state change workflow for **Approval Workflow Executor**.
...
+| `username`                 | Optional           | Username for the external BPMN process engine. This overrides `<ServerUser>` defined in the `api-manager.xml` file for the tenant.                                                                  |
+| `password`                 | Optional           | Password for the external BPMN process engine. This overrides `<ServerPassword>` defined in the `api-manager.xml` file for the tenant.                                                              |
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md:71]: same for username and password

File: en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md [hunk 4]
```diff
@@ -62,8 +62,10 @@ First, enable the API state change workflow for **Approval Workflow Executor**.
 
 The elements of the above configuration are explained below.
 
-| Element name                                       | Description                                                                                                                                               |
-|----------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------|                                                                f                                                                  |
-| `stateList`| Mandatory          | This is a comma separated list of the current state and intended action. For example, Created:Publish,Published:Block 
-
-                                                                |
+| Element Name               | Mandatory/Optional | Description                                                                                                                                                                                         |
```
Comment (shaniR) [inline en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md:65]: we should point to this from the 3rd point where the configuration is done.

**Review by shaniR — DISMISSED** [review]

File: en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md (hunk 1 above)
Comment (Meruja) [reply en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md:67]: These configurations will be used in step 3.

File: en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md (hunk 2 above)
Comment (Meruja) [reply en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md:69]: We need these configurations to enable WS workflow via carbon console as per documentation.

File: en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md (hunk 4 above)
Comment (Meruja) [reply en/docs/learn/design-api/advanced-topics/adding-an-api-state-change-workflow.md:67]: Done with 20c4d8a
Added missing link

**Review by Mariangela — APPROVED** [review]

**Pull request merged**